src/input.rs

```diff
@@ -223,7 +222,7 @@ impl<T> Mapper<T> {
         self.map.insert(action, button);
     }
 
-    pub fn add_actions(&mut self, actions: &[(T, Button)]) // TODO: better solution? don't like & and Copy
```

Comment:
I need to help here. if I don't have `&` and `Copy`, I've got this error:
```
the size for values of type `[(T, input::Button)]` cannot be known at compilation time
the trait `std::marker::Sized` is not implemented for `[(T, input::Button)]`
unsized locals are gated as an unstable feature (rustcE0277)
```
Maybe use macro?

Reply:
Borrowing there with `&` is absolutely fine. You copy data inside of the function. Copy numbers in fact, so you don't need to own data. I suppose we will get mappings from some configuration file or something like that. And most likely it will be in Vec or in HashMap already. So I suggest to keep it as is for now. If we later go with HashMaps, we will make `add_actions` taking ownership of the map.

I've pushed a small commit there. Let me know, if it is not clear.

Reply:
Great! Yes, it's clear